File: baekjoon/1149.py
n = int(input())
arr = [list(map(int, input().split())) for _ in range(n)]

MIN = 21e8

for i in range(3):
    SUM = arr[0][i]
    now = 1
    while now<n:
        MIN = 21e8
        for j in range(3):
            if

File: baekjoon/1149.js
// https://www.acmicpc.net/problem/1149
// RGB거리
// dp
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");

const n = Number(input.shift());
const house = input.map((v) => v.split(" ").map(Number));

function solution() {
  for (let i = 1; i < house.length; i++) {
    house[i][0] += Math.min(house[i - 1][1], house[i - 1][2]);
    house[i][1] += Math.min(house[i - 1][0], house[i - 1][2]);
    house[i][2] += Math.min(house[i - 1][0], house[i - 1][1]);
  }
  console.log(Math.min(...house.pop()));
}

solution();
